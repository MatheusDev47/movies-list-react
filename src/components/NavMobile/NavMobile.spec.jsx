import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { NavMobile } from '.';

describe('<NavMobile />', () => {
  it('Should navbar mobile is render correctly in the page', () => {
    const fn = jest.fn();

    render(<NavMobile onClick={fn} value="X" />);

    expect(screen.getByRole('list')).toBeInTheDocument();
    expect(screen.getAllByRole('link')).toHaveLength(3);
  });

  it('Should onClick function in the button is working', () => {
    const fn = jest.fn();

    render(<NavMobile onClick={fn} value="X" />);

    const button = screen.getByRole('button', { name: /x/i });

    userEvent.click(button);

    expect(button).toBeInTheDocument();
    expect(fn).toHaveBeenCalledTimes(1);
  });

  it('Should match snapshot', () => {
    const fn = jest.fn();

    const { container } = render(<NavMobile onClick={fn} value="X" />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
