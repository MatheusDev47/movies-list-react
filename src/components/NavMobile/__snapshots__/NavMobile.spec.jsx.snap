// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavMobile /> Should match snapshot 1`] = `
<nav
  class="navbar-mobile"
>
  <div
    class="button"
  >
    <button
      class="button-menu-close"
    >
      X
    </button>
  </div>
  <ul>
    <li>
      <a
        href="#"
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="#"
      >
        Popular Movies
      </a>
    </li>
    <li>
      <a
        href="#"
      >
        More Movies
      </a>
    </li>
  </ul>
</nav>
`;
